R/selections.R

```diff
@@ -172,7 +172,12 @@ recipes_eval_select <- function(quos, data, info, ..., allow_rename = FALSE) {
 
   local_current_info(nested_info)
 
-  expr <- expr(c(!!!quos))
+  if (is_quosures(quos)) {
```

Comment:
You have two options:

You could use `enquos(case_weights)` (i.e. the plural version that gives you a list of quosures) so you don't have to special case this in `recipes_eval_select()`.

Ideally you'd use `enquo(case_weights)` like you have, but `recipes_eval_select()` expects a list of quosures so we need to give it one somehow

---

Alternatively in `get_case_weights()` you can probably wrap `selection` in a list like:

```r
wt_col <- unname(recipes_eval_select(list(selection), .data, info))
```

That may be better since you can add a comment here about why you have to wrap it in a list (because `recipes_eval_select()` wants one), but otherwise you can continue to correctly use `enquo(case_weights)`

If all of the case weight supported steps go through `get_case_weights()`, then that sounds good to me